test/support/generator/get-facility-fixed-fee-generator.ts

```diff
@@ -0,0 +1,116 @@
+import { DateString } from '@ukef/helpers';
+import { AcbsGetFacilityFixedFeeResponseDto } from '@ukef/modules/acbs/dto/acbs-get-facility-fixed-fee-response.dto';
+import { DateStringTransformations } from '@ukef/modules/date/date-string.transformations';
+import { GetFacilityFixedFeeResponse } from '@ukef/modules/facility-fixed-fee/dto/get-facility-fixed-fee-response.dto';
+import { FacilityFixedFee } from '@ukef/modules/facility-fixed-fee/facility-fixed-fee.interface';
+
+import { AbstractGenerator } from './abstract-generator';
+import { RandomValueGenerator } from './random-value-generator';
+
+export class GetFacilityFixedFeeGenerator extends AbstractGenerator<FacilityFixedFeeValues, GenerateResult, GenerateOptions> {
+  constructor(protected readonly valueGenerator: RandomValueGenerator, protected readonly dateStringTransformations: DateStringTransformations) {
+    super(valueGenerator);
+  }
+
+  protected generateValues(): FacilityFixedFeeValues {
+    return {
+      fixedFeeAmount: this.valueGenerator.nonnegativeFloat({ fixed: 2 }),
+      effectiveDateInAcbs: this.valueGenerator.dateTimeString(),
+      expirationDateInAcbs: this.valueGenerator.dateTimeString(),
+      nextDueDateInAcbs: this.valueGenerator.dateTimeString(),
+      nextAccrueToDateInAcbs: this.valueGenerator.dateTimeString(),
+      segmentIdentifier: this.valueGenerator.stringOfNumericCharacters({ length: 2 }),
+      description: this.valueGenerator.string(),
+      currency: {
+        currencyCode: this.valueGenerator.string(),
+      },
+      lenderType: {
+        lenderTypeCode: this.valueGenerator.string(),
+      },
+      incomeClass: {
+        incomeClassCode: this.valueGenerator.string({ length: 3 }),
+      },
+      spreadToInvestorsIndicator: this.valueGenerator.boolean(),
+    };
+  }
+
+  protected transformRawValuesToGeneratedValues(
+    values: FacilityFixedFeeValues[],
+    { facilityIdentifier, portfolioIdentifier }: GenerateOptions,
+  ): GenerateResult {
+    const facilityFixedFeesInAcbs: AcbsGetFacilityFixedFeeResponseDto = values.map((v) => ({
```

Comment:
Can we please have a meaningful variable name instead of `v` please.

Reply:
updated